### PR TITLE
Add more unicode_literals to modules, states, and test files

### PR DESCRIPTION
**Pull request by rallytime**

### CODE REVIEW

File: salt/modules/mount.py [hunk 1]
```diff
@@ -291,7 +293,7 @@ def __str__(self):
 
     def __repr__(self):
         '''always works'''
-        return str(self.criteria)
+        return six.text_type(self.criteria)
```
Comment (terminalmage) [inline salt/modules/mount.py]: I think you and @DmitryKuzmenko may be both working on the same file here? I recall talking through this with him and we found that we actually need to change the `str` to `repr`, not `six.text_type`. Or maybe this is just similar code to what he was working on.

File: salt/modules/mount.py [hunk 2]
```diff
@@ -365,7 +367,7 @@ def __str__(self):
...
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same here, `six.text_type` should be `repr`.

File: salt/modules/mount.py [hunk 3]
```diff
@@ -483,15 +487,15 @@ def rm_fstab(name, device, config='/etc/fstab'):
 
     except (IOError, OSError) as exc:
         msg = "Couldn't read from {0}: {1}"
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 4]
```diff
@@ -483,15 +487,15 @@ def rm_fstab(name, device, config='/etc/fstab'):
...
 
     if modified:
         try:
             with salt.utils.files.fopen(config, 'w+') as ofile:
-                ofile.writelines(lines)
+                ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: This needs to be `encode`, not `decode`, and the `fopen()` should have a mode of `'wb'`.

File: salt/modules/mount.py [hunk 5]
```diff
@@ -483,15 +487,15 @@ def rm_fstab(name, device, config='/etc/fstab'):
...
         except (IOError, OSError) as exc:
             msg = "Couldn't write to {0}: {1}"
-            raise CommandExecutionError(msg.format(config, str(exc)))
+            raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 6]
```diff
@@ -617,19 +622,19 @@ def set_fstab(
 
     except (IOError, OSError) as exc:
         msg = 'Couldn\'t read from {0}: {1}'
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 7]
```diff
@@ -617,19 +622,19 @@ def set_fstab(
...
+        lines.append(six.text_type(entry))
         ret = 'new'
 
     if ret != 'present':  # ret in ['new', 'change']:
         if not salt.utils.args.test_mode(test=test, **kwargs):
             try:
                 with salt.utils.files.fopen(config, 'w+') as ofile:
                     # The line was changed, commit it!
-                    ofile.writelines(lines)
+                    ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same as above, this needs to be an `encode`, and we need to open the file using `'wb'` as the mode.

File: salt/modules/mount.py [hunk 8]
```diff
@@ -745,19 +751,19 @@ def set_vfstab(
 
     except (IOError, OSError) as exc:
         msg = 'Couldn\'t read from {0}: {1}'
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 9]
```diff
@@ -745,19 +751,19 @@ def set_vfstab(
...
+        lines.append(six.text_type(entry))
         ret = 'new'
 
     if ret != 'present':  # ret in ['new', 'change']:
         if not salt.utils.args.test_mode(test=test, **kwargs):
             try:
                 with salt.utils.files.fopen(config, 'w+') as ofile:
                     # The line was changed, commit it!
-                    ofile.writelines(lines)
+                    ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same as above, this needs to be an `encode`, and we need to open the file using `'wb'`.

File: salt/modules/mount.py [hunk 10]
```diff
@@ -811,14 +818,14 @@ def rm_automaster(name, device, config='/etc/auto_salt'):
                 lines.append(line)
     except (IOError, OSError) as exc:
         msg = "Couldn't read from {0}: {1}"
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 11]
```diff
@@ -811,14 +818,14 @@ def rm_automaster(name, device, config='/etc/auto_salt'):
...
 
     try:
         with salt.utils.files.fopen(config, 'w+') as ofile:
-            ofile.writelines(lines)
+            ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.

File: salt/modules/mount.py [hunk 12]
```diff
@@ -811,14 +818,14 @@ def rm_automaster(name, device, config='/etc/auto_salt'):
...
     except (IOError, OSError) as exc:
         msg = "Couldn't write to {0}: {1}"
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 13]
```diff
@@ -893,26 +901,26 @@ def set_automaster(
...
                                 name, type_opts, device_fmt)
-                       )
+                        )
                         lines.append(newline)
                 else:
                     lines.append(line)
     except (IOError, OSError) as exc:
         msg = 'Couldn\'t read from {0}: {1}'
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
```
Comment (terminalmage) [inline salt/modules/mount.py]: `six.text_type()` is unnecessary here

File: salt/modules/mount.py [hunk 14]
```diff
@@ -893,26 +901,26 @@ def set_automaster(
...
 
     if change:
         if not salt.utils.args.test_mode(test=test, **kwargs):
             try:
                 with salt.utils.files.fopen(config, 'w+') as ofile:
                     # The line was changed, commit it!
-                    ofile.writelines(lines)
+                    ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.

File: salt/modules/mount.py [hunk 15]
```diff
@@ -934,7 +942,7 @@ def set_automaster(
                 try:
                     with salt.utils.files.fopen(config, 'w+') as ofile:
                         # The line was changed, commit it!
-                        ofile.writelines(lines)
+                        ofile.writelines(salt.utils.data.decode(lines))
```
Comment (terminalmage) [inline salt/modules/mount.py]: Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.

File: salt/states/mongodb_user.py [hunk 16]
```diff
@@ -116,7 +117,7 @@ def present(name,
         #    users= (False, 'not authorized on admin to execute command { usersInfo: "root" }')
         if not users[0]:
             ret['result'] = False
-            ret['comment'] = "Mongo Err: "+str(users[1])
+            ret['comment'] = "Mongo Err: {0}".format(six.text_type(users[1]))
```
Comment (terminalmage) [inline salt/states/mongodb_user.py]: `six.text_type()` is unnecessary here

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: salt/modules/mount.py (hunk 1 above)
Comment (rallytime) [reply salt/modules/mount.py]: hrm...I'm not sure. i'll just fix this here and i can rebase if needed.

File: salt/modules/mattermost.py [hunk 17]
```diff
@@ -133,7 +134,7 @@ def post_message(message,
         parameters['username'] = username
     parameters['text'] = '```' + message + '```'  # pre-formatted, fixed-width text
     log.debug('Parameters: %s', parameters)
-    data = str('payload={0}').format(salt.utils.json.dumps(parameters))  # future lint: disable=blacklisted-function
+    data = six.text_type('payload={0}').format(salt.utils.json.dumps(parameters))  # future lint: disable=blacklisted-function
```
[the remark this answers is not in the document]
Comment (rallytime) [reply salt/modules/mattermost.py]: 👍

**Review by terminalmage — APPROVED** [review]

**Pull request merged**